### PR TITLE
Put the segment import inside a try catch as its not used in controll…

### PR DESCRIPTION
**Pull request by cshiels-ie**
…er image.

---

> [!NOTE]
> Makes Segment dependency optional with graceful no-op when unavailable or missing write_key, switches to per-call anonymous_id tracking, and adds extensive unit tests.
> 
> - **StorageSegment (`metrics_utility/library/storage/segment.py`)**:
>   - Make `segment.analytics` import optional via `SEGMENT_AVAILABLE`; log when unavailable.
>   - Gracefully no-op in `put()` when Segment is unavailable or `write_key` is missing; improve param validation (`dict is None`).
>   - Switch tracking from `user_id` to per-call `anonymous_id` (UUID); configure `analytics` and flush after sends.
>   - Preserve chunking behavior; include chunk metadata and debug output.
> - **Tests (`metrics_utility/test/library/test_storage_segment.py`)**:
>   - Add comprehensive tests covering availability, no `write_key`, default/custom events, chunking/bulk limits, debug output, size/calculation, and various data/edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 45cfd6437e21117ea3c9b2a1170a323f2d298557. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: metrics_utility/library/storage/segment.py [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+
+try:
+    import segment.analytics as analytics
+
+    SEGMENT_AVAILABLE = True
+except ImportError:
+    analytics = None
+    SEGMENT_AVAILABLE = False
+
+logger = logging.getLogger(__name__)
```
Comment (himdel) [inline metrics_utility/library/storage/segment.py]: import logger from metrics_utility.logger for now

otherwise lgtm :)

**Review by himdel — APPROVED** [review]
